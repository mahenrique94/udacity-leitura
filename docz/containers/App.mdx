---
menu: Containers
name: App
route: /
---

import { Playground } from "docz"

import App from "../../src/containers/App"

# App

## Basic usage

<Playground>
    <App>
        <h1>App</h1>
    </App>
</Playground>
