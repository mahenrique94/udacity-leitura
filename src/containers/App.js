import PropTypes from 'prop-types'
import React, { Fragment } from 'react'

import { Container } from 'reactstrap'

import Menu from 'components/menu/Menu'
import Notifications from 'components/notifications/Notifications'

const App = ({ children }) => (

    <Fragment>
        <Notifications/>
        <Menu/>
        <Container className="mb-5">
            { children }
        </Container>
    </Fragment>

)

App.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.array,
        PropTypes.object,
        PropTypes.string
    ]).isRequired
}

export default App
