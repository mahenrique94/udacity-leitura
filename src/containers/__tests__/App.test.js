import React from "react"
import renderer from "react-test-renderer"

import App from "../App"

describe("Tests for App container", () => {

    it("Should rendering without crash", () => {
        const tree = renderer.create(
            <App>
                <h1>App</h1>
            </App>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

})
