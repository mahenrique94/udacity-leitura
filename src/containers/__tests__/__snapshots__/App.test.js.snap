// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for App container Should rendering without crash 1`] = `
<h1>
  App
</h1>
`;
